Source/Fuse.Models/ModelJavaScript.uno

```diff
@@ -0,0 +1,165 @@
+using Uno;
+using Uno.UX;
+using Uno.Collections;
+using Uno.Compiler;
+using Uno.IO;
+using Uno.Text;
+using Fuse;
+using Fuse.Reactive;
+using Fuse.Scripting;
+
+namespace Fuse.Models
+{
+	public class ModelJavaScript : JavaScript, IPreviewStateSaver
+	{
+		class ModelData
+		{
+			public string ModulePath;
+			public NameTable NameTable;
+		}
+		static PropertyHandle _modelHandle = Properties.CreateHandle();
+		
+		//Requires a NameTable that will be set first.
+		[UXAttachedPropertySetter("JavaScript.Model"), UXAuxNameTable("ModelNameTable")]
+		public static void SetModel(Visual v, string modulePath)
+		{
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData { ModulePath = modulePath };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.ModulePath = modulePath;
+			}
+			
+			Complete( md, v );
+		}
+		
+		static void Complete( ModelData md, Visual v )
+		{
+			v.RemoveAllChildren<ModelJavaScript>();
+			
+			//avoid creating without the NameTable as unfortunately UX will set the Model prior to the NameTable
+			if (md.NameTable == null || md.ModulePath == null)
+				return;
+			
+			v.Children.Add( new ModelJavaScript(md.NameTable, md.ModulePath, null) );
+		}
+		
+		[UXAttachedPropertySetter("ModelNameTable")]
+		public static void SetModelNameTable(Visual v, NameTable nt)
+		{
+			var md = v.Properties.Get( _modelHandle ) as ModelData;
+			if (md == null)
+			{
+				md = new ModelData{ NameTable = nt };
+				v.Properties.Set( _modelHandle, md );
+			}
+			else
+			{
+				md.NameTable = nt;
+			}
+			
+			Complete( md, v );
+		}
+
+		//TODO: This should probably be JavaScript.Model, Preview would need to be adjusted as well
+		[UXAttachedPropertySetter("Model")]
```

Comment:
Ticket and link, please?

Reply:
@mortoray I've tried changing this to `ModelJavaScript.Model`, but the UX compiler complains that the `Model` attribute could not be found. I suppose it's illegal to have multiple attached properties with the same qualified name, even if they can be discriminated by the type of their first argument? In any case, I assume the comment about preview is outdated (went away with the introduction of IRootVisualProvider)

Reply:
I thought that Preview needed to know about `Model`. But if not then we don't have to worry about that part.

I'd still call it `JavaScript.Model` though, since `ModelJavaScript` is an internal detail.  I don't know why the UX compiler would give a conflict though.

Reply:
@mortoray Unless you know of any more places, [this is the only place I know of where preview used to have knowledge of the model feature](https://github.com/fusetools/Fuse/commit/03067734d22ecf54330297a0e9988b3e142581d5#diff-7eae56ceee21977d27ede24e1ae9569dL61). The rest has been abstracted away via `IRootVisualProvider` and your preview state system IIRC.

Reply:
Nope. That was it, thus it should be safe to change.